### PR TITLE
Refactor: eligibility forms

### LINKED ISSUE #1564: Refactor Eligibility form fields away from model
In the same vein as #1442, we can think about simplifying how configuration for the eligibility verification forms is moved out to the front-end.

Today we model very specific fields and information on the `EligibilityVerifier` class. This leads to a bloated model definition and plenty of logic just to copy configuration from the model into a form instance.

Instead, the model could reference a form class by name, and all the configuration can live directly within each form class where it is needed. The model could facilitate instantiating an instance of its own form class as a helper.

## Acceptance Criteria

- [ ] Each of the existing `EligibilityVerifier.form_*` fields is removed
- [ ] A new field `EligibilityVerifier.form` captures the full module path for a form class like `benefits.eligibility.forms.FormClass`
- [ ] Configuration for the forms lives directly within a form class
- [ ] `EligibilityVerifier` can produce an instance of its form using something like https://stackoverflow.com/a/30941292/453168

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #1564

Gets us closer to #1525

## What this PR does

* Removes all the form-specific fields from `EligibilityVerifier` model: `form_blurb`, `form_sub_label`, etc.
* Adds a single field `form_class` that holds the name of a Python form class, adds a method to create an instance of this class
* Creates a form class for `MSTCourtesyCard` verification, moves all config there (out of migration)

## Testing

* Run `bin/init.sh`
* Verify with MST Courtesy Card flow
* Ensure no functional regressions
* Ensure no design regressions
* Ensure no copy regressions

### CODE REVIEW

**Comment by machikoyasuda** [comment]
Reviewing now

File: benefits/eligibility/forms.py [hunk 1]
```diff
@@ -53,32 +53,74 @@ class EligibilityVerificationForm(forms.Form):
...
+        name_label,
+        name_placeholder,
+        name_help_text,
+        sub_label,
+        sub_placeholder,
+        sub_help_text,
+        name_max_length=None,
+        sub_input_mode=None,
+        sub_max_length=None,
+        sub_pattern=None,
```
Comment (machikoyasuda) [inline benefits/eligibility/forms.py:70]: These comments from the data file could become docstring or comments on this file to describe these:

```python
    # A regular expression used to validate the 'sub' API field before sending to this verifier
    form_sub_pattern = models.TextField(null=True)
    # Input mode can be "numeric", "tel", "search", etc. to override default "text" keyboard on mobile devices
    form_input_mode = models.TextField(null=True)
    # The maximum length accepted for the 'sub' API field before sending to this verifier
    form_max_length = models.PositiveSmallIntegerField(null=True)
    # The maximum length accepted for the 'name' API field before sending to this verifier
    form_name_max_length = models.PositiveSmallIntegerField(null=True)
```

**Review by machikoyasuda — DISMISSED** [review]
Tested:
- Desktop design: `col-lg-6` is correct ✅ 
- English MST CC flow / Good user, bad user - Desktop ✅ 
- Spanish MST CC flow / Good user, bad user - Desktop ✅ 
- English MST CC flow / Good user, bad user - iPhone simulator - Number keypad appears properly ✅ 
image

**Comment by angela-tran** [comment]
👀

File: benefits/eligibility/forms.py (hunk 1 above)
Comment (thekaveman) [reply benefits/eligibility/forms.py:70]: Thanks for pasting here 🙏 I'll add those.

Comment (thekaveman) [reply benefits/eligibility/forms.py:70]: [dae5371](https://github.com/cal-itp/benefits/pull/1657/commits/dae53710d5769c7248f4d050c4c2682fcbca5631)

Comment (machikoyasuda) [reply benefits/eligibility/forms.py:70]: 🎉🎉🎉

**Review by machikoyasuda — APPROVED** [review]

File: benefits/eligibility/forms.py (hunk 1 above)
Comment (thekaveman) [reply benefits/eligibility/forms.py:70]: Ahh, typo... one more rebase incoming

**Review by angela-tran — APPROVED** [review]

**Comment by thekaveman** [comment]
Merging based on @machikoyasuda's prior approval. Thanks!

**Pull request merged**